test/support.js

```diff
@@ -30,9 +30,6 @@ beforeEach(function() {
       {type: 'STUB'}
     ]
   });
-
-  // auto attach data sources to models
-  loopback.autoAttach();
```

Comment:
> Do you know the answer yet? ;-)

Yes I got it :). Its very odd that attaching to a `dataSource` modifies your property definitions :(.